appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/interceptors/CircuitBreakerInterceptor.java

```diff
@@ -214,10 +214,15 @@ private Object circuitBreak(InvocationContext invocationContext) throws Exceptio
         long delay = (Long) FaultToleranceCdiUtils.getOverrideValue(
                 config, CircuitBreaker.class, "delay", invocationContext, Long.class)
                 .orElse(circuitBreaker.delay());
-        // Look for a String and cast to ChronoUnit - Use the Common Sense Convertor
-        ChronoUnit delayUnit = (ChronoUnit) FaultToleranceCdiUtils.getOverrideValue(
-                config, CircuitBreaker.class, "delayUnit", invocationContext, String.class)
-                .orElse(circuitBreaker.delayUnit());
+
+        ChronoUnit delayUnit = circuitBreaker.delayUnit();
+        // Check for an override
+        Optional<String> delayUnitOverride = FaultToleranceCdiUtils.getOverrideValue(
```

Comment:
Why all this complex code? I think it's enough to use ChronoUnit.class instead of String.class as a parameter to `getOverrideValue`:

```
ChronoUnit delayUnit = (ChronoUnit) FaultToleranceCdiUtils.getOverrideValue(
                config, CircuitBreaker.class, "delayUnit", invocationContext, ChronoUnit.class)
                .orElse(circuitBreaker.delayUnit());
```

It would call config.getOptionalValue(..., ChronoUnit.class) which is valid because Config has a default converter for enums (in general, for all classes that provide static valueOf(String) method).

Reply:
I'm fairly certain that it used to be that, but then config changed and I had to use string. I was lead to believe that this wasn't a bug and was intentional. I'll check when I have some time

Reply:
You were right, maybe I misremembered. Changes made :)